src/Atmos/Model/source.jl

```diff
@@ -38,33 +38,36 @@ struct GeostrophicForcing{FT} <: Source
   u_geostrophic::FT
   v_geostrophic::FT
 end
-function atmos_source!(s::GeostrophicForcing, m::AtmosModel, source::Vars, state::Vars, aux::Vars, t::Real)
-  u = state.ρu / state.ρ
+function atmos_source!(s::GeostrophicForcing, atmos::AtmosModel, source::Vars, state::Vars, aux::Vars, t::Real)
   u_geo = SVector(s.u_geostrophic, s.v_geostrophic, 0)
-  source.ρu -= state.ρ * s.f_coriolis * (u - u_geo)
+  ẑ = vertical_unit_vector(atmos.orientation, aux)
+  fkvector = s.f_coriolis * ẑ
+  source.ρu -= fkvector × (state.ρu .- state.ρ*u_geo)
```

Comment:
Need to confirm @smarras79's question on sign here.

Reply:
The sign is [correct](https://github.com/climate-machine/CLIMA/pull/619#discussion_r366111032).